Sources/OpenAPIRuntime/Interface/ClientTransport.swift

```diff
@@ -122,14 +132,16 @@ public protocol ClientTransport: Sendable {
     /// HTTP response.
     /// - Parameters:
     ///   - request: An HTTP request.
+    ///   - body: An HTTP request body.
     ///   - baseURL: A server base URL.
     ///   - operationID: The identifier of the OpenAPI operation.
-    /// - Returns: An HTTP response.
+    /// - Returns: An HTTP response and its body.
     func send(
-        _ request: Request,
+        _ request: HTTPRequest,
+        body: HTTPBody?,
         baseURL: URL,
         operationID: String
-    ) async throws -> Response
+    ) async throws -> (HTTPResponse, HTTPBody?)
```

Comment:
It could, I just haven't seen that used much in Apple public APIs: https://github.com/search?q=org%3Aapple+content%3A%2F-%3E+%5C%28%5B%5Cw%5Cd%5D%2B%3A+%5B%5Cw%5Cd%5D%2B%5B%2C%5C%29%5D%2F+path%3A*.swift&type=code

From me skimming the above, we only use labels where the type itself could be ambiguous (e.g. two returned `Int`s) or isn't expressive enough (e.g. `(path: String)`). Here both the response and body are first class types, so I don't think there'd be any confusion, and in those cases I don't see much labels used just for easier access.

FWIW, I haven't even thought about this before you just brought it up, because in my prototyping, I always did the following:

```swift
let (response, responseBody) = client.send(...)
```

I imagine folks will do that, or will just forward both as a single value to the caller, e.g. in middlewares:

```swift
return try await next(...) // the returned value of both `next` and the method itself is (HTTPResponse, HTTPBody?)
```

Reply:
Right, and there it makes sense, because the types are `(Bool, Element)`, which could reasonably be interpreted many ways.

We're returning just `(HTTPResponse, HTTPBody?)`, which I think is pretty unambiguous.

Reply:
I wasn't suggesting it was ambiguous when you look at the types. I was suggesting it helps the call site for those that access the tuple members—this change has no impact on users that don't use `.0` and `.1`, but does greatly improve the experience for those that do.

Reply:
I also realized the exact shape of the API was the subject of SOAR-0005, so I'm a little uneasy about changing them now, as that becomes part of the public API.

Reply:
🙂 OK.